File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.java
class Solution {
    public int missingInteger(int[] nums) {
        int c=0;
        
        Set<Integer> him=new HashSet<>();
        for(int i:nums){
            him.add(i);
        }
        c=nums[0];
        
        for(int i=1;i<nums.length;i++){
            if(nums[i]!=(nums[i-1]+1)){
                break;
            }
            c+=nums[i];
            
        }
        int x=c;
        while(him.contains(x)){
            x++;
        }
        return x;

    }
}

File: 2996-smallest-missing-integer-greater-than-sequential-prefix-sum/2996-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        c=0
        v=set()
        for i in nums:
            v.add(i)
        
        c=nums[0]
        for i in range(1,len(nums)):
            if((nums[i]-1 )!=nums[i-1]):
                break
            c+=nums[i]
        x=c
        print(c)
        while x in v:
            x+=1
        return x